rundeckapp/grails-app/services/rundeck/services/ExecutionService.groovy

```diff
@@ -2410,14 +2433,42 @@ class ExecutionService implements ApplicationContextAware, StepExecutor, NodeSte
      * @param props
      * @param scheduledExec
      * @param authContext auth for reading storage defaults
+     * @param optionValues values modified by a job plugin
      * @return
      */
     private HashMap validateJobInputOptions(Map props, ScheduledExecution scheduledExec, UserAndRolesAuthContext authContext) {
-        HashMap optparams = parseJobOptionInput(props, scheduledExec, authContext)
+        HashMap optparams
+        optparams = parseJobOptionInput(props, scheduledExec)
+        def result = checkBeforeJobExecution(scheduledExec, optparams, props, authContext)
+        if(result?.useNewValues()){
+            optparams = result.optionsValues
+            checkSecuredOptions(scheduledExec, props, result.optionsValues)
+        }
         validateOptionValues(scheduledExec, optparams,authContext)
         return optparams
     }
 
+    def checkSecuredOptions(scheduledExecution, props = [:], optionsValues){
```

Comment:
I'm not sure what this method is supposed to do.  I see it is called in `scheduleAdHocJob` and `retryExecuteJob` but i don't know why.

Reply:
This is to replace values comming from job secure options (by the ones setted up by a JobLifeCyclePlugin)